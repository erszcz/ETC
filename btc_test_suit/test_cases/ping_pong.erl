-module(ping_pong).
-compile(export_all).

-type protocol() :: {ping, pid()} | {pong, pid()}.

pingClient(ServerPid) ->
    ServerPid ! {ping, self()},
    receive
        {pong,_} ->
            "Done"
    end.

pongServer() ->
    receive
        {ping, Ping_PID} ->
            Ping_PID ! {pong, self()},
            pongServer()
    end.

start()->
    ServerPid = spawn(fun pongServer/0),
    spawn(fun () -> pingClient(ServerPid) end),
    ok.