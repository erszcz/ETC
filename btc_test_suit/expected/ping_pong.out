Module ping_pong: 
  {pingClient,1} :: [Padd A;].(A)->[char]
  {pongServer,0} :: ()->A
  {start,0} :: ()->atom

test_cases/ping_pong.erl:2: Warning: export_all flag enabled - all functions will be exported
test_cases/ping_pong.erl:4: Warning: type protocol() is unused
